File: backend_project/src/main/java/com/ssafy/controller/room/group/GroupDetailResponse.java
package com.ssafy.controller.room.group;

import lombok.*;

import java.util.List;

// 특정 조 상세 정보
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GroupDetailResponse {

    private int groupNo;
    private double progress;
    private List<MemberDto> members;
    private List<VerificationPhotoDto> verificationPhotos;
    private List<VisitedPlaceDto> visitedPlaces;

    // 조원 정보
    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class MemberDto {
        private long userId;
        private String nickname;
        private String codeId;  // "J001"=팀장, "J002"=팀원
    }

    // 인증샷
    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class VerificationPhotoDto {
        private long pictureId;
        private String pictureUrl;
        private long missionId;
        private String completionTime; // "2025-02-12 10:00:00"
    }

    // 방문한 장소
    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class VisitedPlaceDto {
        private long missionId;
        private String positionName;
        private String completedAt; // "yyyy-MM-dd HH:mm:ss"
    }

}

File: TMBG/app/src/main/java/com/ssafy/tmbg/data/team/dao/GroupDetailResponse.kt
package com.ssafy.tmbg.data.team.dao

// Team 데이터
data class GroupDetailResponse(
    val groupNo: Int,
    val progress: Long,               // (완료 미션 수 / 전체 미션 수)
    val members: List<MemberDto>,        // 조에 속한 멤버 목록
    val verificationPhotos: List<VerificationPhotos>, // 조장이 올린 인증샷
    val visitedPlaces: List<VisitedPlace>,
)

data class MemberDto(
    val userId: Long,
    val nickname: String,
    val codeId: String,
)

data class VerificationPhotos(
    val pictureId: Long,
    val pictureUrl: String,
    val missionId: Long,
    val completionTime: String
)

data class VisitedPlace(
    val missionId: Long,
    val positionName: String,
    val completedAt: String
)